### ISSUE TITLE
Couchdb containers always in CrashLoopBackOff status

### ISSUE THREAD

**Issue by yoanisgil** [helm/charts#4666]
**Is this a request for help?**:
Yes
---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
BUG REPORT

**Version of Helm and Kubernetes**:
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.5", GitCommit:"f01a2bf98249a4db383560443a59bed0c13575df", GitTreeState:"clean", BuildDate:"2018-03-19T19:38:17Z", GoVersion:"go1.9.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"8+", GitVersion:"v1.8.9-gke.1", GitCommit:"c205eb8319d25aa46b1eb39b6c1d2bc5d1a49fae", GitTreeState:"clean", BuildDate:"2018-03-13T17:49:52Z", GoVersion:"go1.8.3b4", Compiler:"gc", Platform:"linux/amd64"}
```

```
$ helm version
Client: &version.Version{SemVer:"v2.8.2", GitCommit:"a80231648a1473929271764b920a8e346f6de844", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.8.2", GitCommit:"a80231648a1473929271764b920a8e346f6de844", GitTreeState:"clean"}
```
**Which chart**:

couchdb

**What happened**:

I tried to install couchdb by running `helm install --name tti  incubator/couchdb --set allowAdmdinParty=true` and after a few seconds I see this:

```
NAME                   READY     STATUS             RESTARTS   AGE
my-release-couchdb-0   1/2       CrashLoopBackOff   2          37s
my-release-couchdb-1   1/2       CrashLoopBackOff   2          37s
my-release-couchdb-2   1/2       CrashLoopBackOff   2          37s
```

which clearly indicates that the pods are crashing.

**What you expected to happen**:

I would have expected to have 3 running and healthy couchdb containers.

**How to reproduce it** (as minimally and precisely as possible):

This k8s cluster is powered by GKE.

**Anything else we need to know**:

When I run this command `kubectl logs -f my-release-couchdb-0 couchdb` I see this:

```
chown: changing ownership of '/opt/couchdb/etc/default.d/..data': Read-only file system
chown: changing ownership of '/opt/couchdb/etc/default.d/chart.ini': Read-only file system
chown: changing ownership of '/opt/couchdb/etc/default.d/..2018_04_03_19_22_05.801040662/chart.ini': Read-only file system
chown: changing ownership of '/opt/couchdb/etc/default.d/..2018_04_03_19_22_05.801040662': Read-only file system
chown: changing ownership of '/opt/couchdb/etc/default.d': Read-only file system
```

**Comment by yoanisgil** [comment]
I would say the issue here is that couchdb fails to boot because the filesystem is read-only? I tried using a persistent volume but that didn't seem to help.

**Comment by emsixteeen** [comment]
Looks to me that the root cause is this: 

https://github.com/apache/couchdb-docker/blob/master/2.1.1/docker-entrypoint.sh#L29

combined with:

https://github.com/kubernetes/charts/blob/master/incubator/couchdb/templates/statefulset.yaml#L68

Since Kubernets v1.9.4 configMaps are mounted as read-only. Quoting from the v1.9.4 [changelog](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.9.md#other-notable-changes-2): 

"Changes secret, configMap, downwardAPI and projected volumes to mount read-only, instead of allowing applications to write data and then reverting it automatically. Until version 1.11, setting the feature gate ReadOnlyAPIDataVolumes=false will preserve the old behavior. (#58720, @joelsmith)"

The CouchDB 2.11 docker image attempts to execute `chown -R couchdb:couchdb /opt/couchdb`, but the directory `/opt/couchdb/etc/default.d` is read-only!

**Comment by yoanisgil** [comment]
@emsixteeen that makes sense, though I was using kubernetes 1.8.9:
```
Server Version: version.Info{Major:"1", Minor:"8+", GitVersion:"v1.8.9-gke.1", GitCommit:"c205eb8319d25aa46b1eb39b6c1d2bc5d1a49fae", GitTreeState:"clean", BuildDate:"2018-03-13T17:49:52Z", GoVersion:"go1.8.3b4", Compiler:"gc", Platform:"linux/amd64"}
```

I can give it a try once the fix is merged.

**Comment by emsixteeen** [comment]
err, my bad -- I saw 1.9.5, but looking closer -- that was for kubectl :) ... either way, after applying that fix, I had no issues starting it up ...

**Comment by kocolosk** [comment]
A similar fix was merged in #5440
